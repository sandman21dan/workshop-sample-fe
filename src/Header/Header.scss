.nav {
  $nav-color: #6464e4;

  height: 4rem;
  background-color: $nav-color;
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 0 3rem;

  &__logo {
    $logo-size: 2.2rem;
    $logo-colour: orange;
    $logo-offset: .8rem;

    @mixin logoBase() {
      width: $logo-size;
      height: $logo-size;
      border-radius: 50%;
    }

    @include logoBase();
    background-color: darken($logo-colour, 5%);
    position: relative;

    &::before {
      content: ' ';
      position: absolute;
      @include logoBase();
      background-color: $logo-colour;
      right: -$logo-offset;
    }

    &::after {
      content: ' ';
      position: absolute;
      @include logoBase();
      background-color: lighten($logo-colour, 5%);
      right: -(2 * $logo-offset);
    }
  }
}
