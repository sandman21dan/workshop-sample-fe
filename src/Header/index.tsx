import * as React from 'react';

import './Header.scss';

const Header: React.FunctionComponent = () => (
  <nav className="nav">
    <div className="nav__logo"></div>
  </nav>
);

export default Header;
