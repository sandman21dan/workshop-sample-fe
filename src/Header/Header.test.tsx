import * as React from 'react';

import Header from './';
import { shallow } from 'enzyme';

it('render correctly', () => {
  const component = shallow(<Header />);

  expect(component).toMatchSnapshot();
});
