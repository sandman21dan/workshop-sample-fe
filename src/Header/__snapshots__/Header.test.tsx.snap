// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<nav
  className="nav"
>
  <div
    className="nav__logo"
  />
</nav>
`;
